plutus-ir/test/Spec.hs

```diff
@@ -28,15 +29,31 @@ import qualified Data.Text.Prettyprint.Doc            as PP
 main :: IO ()
 main = defaultMain $ runTestNestedIn ["test"] tests
 
-instance GetProgram (Term TyName Name ()) where
-    getProgram = trivialProgram . compile
+instance GetProgram (Quote (Term TyName Name ())) where
+    getProgram = trivialProgram . compileAndMaybeTypecheck False
 
 goldenPir :: String -> Term TyName Name () -> TestNested
 goldenPir name value = nestedGoldenVsDoc name $ prettyDef value
 
-compile :: Term TyName Name () -> PLC.Term TyName Name ()
-compile pir = case runCompiling $ compileTerm pir of
-    Right plc -> plc
+typecheck :: PLC.Term TyName Name () -> Either (PLC.Error ()) (PLC.Term TyName Name ())
+typecheck t = runQuoteT $ do
+    renamed <- PLC.rename t
+    _ <- PLC.typecheckTerm (PLC.TypeCheckCfg PLC.defaultTypecheckerGas $ PLC.TypeConfig True mempty) =<< PLC.annotateTerm renamed
+    pure renamed
+
+rename :: PLC.Term TyName Name () -> Either (PLC.Error ()) (PLC.Term TyName Name ())
+rename t = runQuoteT $ PLC.rename t
```

Comment:
This can be puzzling for people reading code. I mean, we have two `rename` functions and one of them executes `runQuote` and the other one doesn't. Maybe `runRename` at least? How often is this function used? Can it be inlined, so that it's clear where `runQuote` is happening? We both did mistakes with it, so I think being explicit here is a good strategy.

Reply:
Sure, I guess. This is only really here so I have something to use when I'm not using the typechecking function below!